lib/gtf/gz_reader.cc

```diff
@@ -0,0 +1,75 @@
+/*
+Author: Hossein Asghari
+*/
+
+#include <string>
+#include <zlib.h>
+#include "gz_reader.h"
+
+gz_reader::gz_reader()
+{
+    gzinput = Z_NULL;
+    zbuffer = (char *) malloc(BUFFSIZE);
+
+    buff_pos = 0;
+    buff_size = 0;
+}
+
+gz_reader::~gz_reader() 
+{
+    free(zbuffer);
+}
+
+int gz_reader::open_gz(const string &file) 
+{
+    gzinput = gzopen(file.c_str(), "r");
+    
+    if (gzinput == Z_NULL) {
+        printf("Error: Could not open gzip file %s\n", file.c_str());
+        return 1;
+    }
+
+    return 0;
+}
+
+void gz_reader::close_gz() 
+{
+    if (gzinput != Z_NULL) {
+        gzclose(gzinput);
+    }
+}
+
+void gz_reader::read_buffer(uint32_t size) {
+    buff_size = gzread(gzinput, zbuffer, size);
+    buff_pos = 0;
+
+    if (buff_size == 0 and gzeof(gzinput) == 0) {
+        buff_size = -1;
+    }
+    if (buff_size < 0) {
+        int err;
```

Comment:
I think we should `throw std::runtime_error()` here, not call `exit()`. Calling `exit` is not nice within a library/object. And it is a bad idea in C++ (it exits the process directly, without calling destructor of the C++ objects on the stack or giving a chance to catch the error in anyway).